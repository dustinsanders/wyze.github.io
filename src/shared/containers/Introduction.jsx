// @flow

import type { Style, VNode } from '../../types'
import { css, media } from 'glamor'
import { small } from '../styles/breakpoints'
import { thin } from '../styles'
import Box from '../components/Box'
import H1 from '../components/H1'
import Image from '../components/Image'
import Section from '../components/Section'
import Strong from '../components/Strong'
import preact from 'preact'

const image: Style = css(
  {
    flexBasis: '100%',
    paddingBottom: '1em',
    textAlign: 'center',
  },
  media(small, {
    flexBasis: 'auto',
    paddingBottom: 0,
    textAlign: 'inherit',
  }),
)
const section: Style = media(small, {
  flex: '1 1 75%',
})
const width: Style = css({
  width: '100%',
})

const Introduction = (): VNode => (
  <Box wrap>
    <Section center={false} styles={image}>
      <Image circle alt="Neil Kistner" ext="png" name="me" />
    </Section>
    <Section styles={css(section, width)}>
      <H1 styles={thin}>
        Hello, I&apos;m <Strong text="Neil Kistner" />,
        a software engineer in <Strong text="St. Louis" />.
      </H1>
    </Section>
  </Box>
)

export default Introduction
