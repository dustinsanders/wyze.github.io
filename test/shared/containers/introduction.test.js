// @flow

import Introduction from '../../../src/shared/containers/Introduction'
import snapshot from '../../helpers/snapshot'

describe('<Introduction />', () => {
  it('renders', snapshot.bind(null, Introduction))
})
