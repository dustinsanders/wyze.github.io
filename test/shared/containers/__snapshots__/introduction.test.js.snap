// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Introduction /> renders 1`] = `
"<Box wrap>
  <Section styles=\\"css-1h4yi9h\\">
    <Image circle alt=\\"Neil Kistner\\" ext=\\"png\\" name=\\"me\\"></Image>
  </Section>
  <Section styles=\\"css-3n7ad4\\">
    <H1 styles=\\"css-1wdk04o\\">
      Hello, I'm 
      <Strong text=\\"Neil Kistner\\"></Strong>
      , a software engineer in 
      <Strong text=\\"St. Louis\\"></Strong>
      .
    </H1>
  </Section>
</Box>"
`;
